### DISCUSSION TITLE
How can multiple different items share the same cooldown?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#863] [Q&A] [answered]
How can multiple different items share the same cooldown?

E.g. Potion A and Potion B (plus 300+ other different potions) all share the same 30-60 second cooldown. This prevents having many different potion types and circumventing the cooldown of one potion by using a slightly different potion. Is this possible within the Flare engine currently (also for different types of items)? in World of Warcraft this is called a Global Cooldown and some item types / groups share a cooldown with one another and there are different sets / pools of global cooldowns such as potions and trinkets etc. This makes it so that some items types can be used one after the other but not all. How this plays out can look like this: In 2-3 seconds Potion -> Potion is not possible for they share the same global cooldown. Yet potion -> trinket -> inventory use item 'clicky' is possible to all us in a 3 second window and this is quite powerful and forces the players to explore / play different parts of the game to increase power (if the game design is not trash XD). Although you can see why this mechanic of 'shared cooldowns for common item types' (e.g. all potions or scrolls etc.) can have arbitrarily defined global cooldown 'pools' so that only certain different kinds of item can be used one after the other; this is quite a powerful tool for game development to create emergent player gameplay / progression goals.

What could multiple different items sharing the same cooldown 'pool' look like in the Flare engine?
(Shot in the dark idea: Perhaps 'Potion A' applies a time limited (~30-60 seconds) status effect / debuff (e.g.'Potion Sickness') and some items (such as 'Potion B') do not work when this status effect / debuff is effecting / on the player? Some kind of 'requires_not_X on an item use?)

**Comment by dorkster** [comment, marked as answer]
All cooldowns are currently limited to being per-power, so there's no way to have one power's cooldown affect other powers.
